File: src/main/java/com/jozufozu/exnihiloomnia/common/items/ItemBaseTool.java
package com.jozufozu.exnihiloomnia.common.items;

import com.google.common.collect.Sets;
import com.jozufozu.exnihiloomnia.ExNihilo;
import net.minecraft.block.Block;
import net.minecraft.item.ToolItem;
import net.minecraft.util.ResourceLocation;

import java.util.Set;

public class ItemBaseTool extends ToolItem
{
    public ItemBaseTool(ResourceLocation registryName, ToolMaterial toolMaterial)
    {
        this(registryName, toolMaterial, Sets.newHashSet());
    }
    
    public ItemBaseTool(ResourceLocation registryName, ToolMaterial toolMaterial, Set<Block> effectiveBlocksIn)
    {
        this(registryName, 0, 0, toolMaterial, effectiveBlocksIn);
    }
    
    public ItemBaseTool(ResourceLocation registryName, float damage, float attackSpeed, ToolMaterial toolMaterial, Set<Block> effectiveBlocksIn)
    {
        super(damage, attackSpeed, toolMaterial, effectiveBlocksIn);
        this.setRegistryName(registryName);
        this.setUnlocalizedName(ExNihilo.MODID + "." + registryName.getResourcePath());
        this.setCreativeTab(ExNihiloTabs.ITEMS);
        
        if (ExNihiloItems.hasRegisteredItems())
        {
            ExNihilo.log.warn("Item '" + registryName + "' created after initialization!");
        }
        
        ExNihiloItems.modItems.add(this);
    }
}

File: src/main/kotlin/com/jozufozu/exnihiloomnia/common/items/ItemBaseTool.kt
package com.jozufozu.exnihiloomnia.common.items

import com.google.common.collect.Sets
import com.jozufozu.exnihiloomnia.common.util.Identifiable
import net.minecraft.block.Block
import net.minecraft.item.IItemTier
import net.minecraft.item.Item
import net.minecraft.item.ToolItem
import net.minecraft.util.ResourceLocation

open class ItemBaseTool(final override val identifier: ResourceLocation, toolMaterial: IItemTier, properties: Properties, damage: Float = 0f, attackSpeed: Float = 0f, effectiveBlocksIn: Set<Block> = Sets.newHashSet()) : ToolItem(damage, attackSpeed, toolMaterial, effectiveBlocksIn, properties.group(ExNihiloTabs.ITEMS)), Identifiable<Item> {

    init {
        this.registryName = identifier
    }
}
